RecoEgamma/EgammaIsolationAlgos/src/EleTkIsolFromCands.cc

```diff
@@ -133,20 +157,11 @@ EleTkIsolFromCands::PIDVeto EleTkIsolFromCands::pidVetoFromStr(const std::string
   }
 }
 
-bool EleTkIsolFromCands::passTrackPreselection(const reco::TrackBase& trk, const TrkCuts& cuts) {
+bool EleTkIsolFromCands::passTrackPreselection(const reco::TrackBase& trk, double trackPt, const TrkCuts& cuts) {
   return trk.hitPattern().numberOfValidHits() >= cuts.minHits &&
          trk.hitPattern().numberOfValidPixelHits() >= cuts.minPixelHits &&
-         (trk.ptError() / trk.pt() < cuts.maxDPtPt || cuts.maxDPtPt < 0) && passQual(trk, cuts.allowedQualities) &&
-         passAlgo(trk, cuts.algosToReject) && trk.pt() > cuts.minPt;
-}
-
-bool EleTkIsolFromCands::passMatchingToElectron(
-    SimpleTrack const& trk, const TrkCuts& cuts, double eleEta, double elePhi, double eleVZ) {
-  const float dR2 = reco::deltaR2(eleEta, elePhi, trk.eta, trk.phi);
-  const float dEta = trk.eta - eleEta;
-  const float dZ = eleVZ - trk.vz;
-
-  return dR2 >= cuts.minDR2 && dR2 <= cuts.maxDR2 && std::abs(dEta) >= cuts.minDEta && std::abs(dZ) < cuts.maxDZ;
+         (trk.ptError() / trackPt < cuts.maxDPtPt || cuts.maxDPtPt < 0) && passQual(trk, cuts.allowedQualities) &&
```

Comment:
resort alll those comparison in order of decreasing filtering power balanced by increasing computational complexity
(in practice move ```(trk.ptError() / trackPt < cuts.maxDPtPt || cuts.maxDPtPt < 0)``` to the end AND even this one please ```cuts.maxDPtPt < 0 || trk.ptError() / trackPt < cuts.maxDPtPt`` and then consider just to cut on
trk.qoverpError()*trk.p()  (is not 100% backward compatible, but equivalent and even superior)
(and I never understood why reco::Track does not store what actually fitted....)

Reply:
Thanks for the comment and sharing your opinion about `ptError()`. Absolutely, I should reorder the checks!

I think for the moment we better stick with ```trk.ptError() / trackPt < cuts.maxDPtPt``` to not risk reconstruction differences in this already large PR, if this is OK for you. But I will definitely keep your suggestion in mind!

Just as a reminder: this loop we are commenting on here is not the hot nNtrack^2 loop that had to be optimized. While I understand that obvious optimizations should be done anyway, like the reordering, I don't necessarily think that the cost of this loop justifies an optimization that alters the reco output at this point.